extensions/charts/cisbenchmark.go

```diff
@@ -0,0 +1,298 @@
+		if err != nil {
+			return err
+		}
+
+		watchAppInterface, err = catalogClient.Apps(CISBenchmarkNamespace).Watch(context.TODO(), metav1.ListOptions{
+			FieldSelector:  "metadata.name=" + CISBenchmarkCRDName,
+			TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+		})
+		if err != nil {
+			return err
+		}
+
+		err = wait.WatchWait(watchAppInterface, func(event watch.Event) (ready bool, err error) {
+			chart := event.Object.(*catalogv1.App)
+			if event.Type == watch.Error {
+				return false, fmt.Errorf("there was an error uninstalling CIS benchmark chart")
+			} else if event.Type == watch.Deleted {
+				return true, nil
+			} else if chart == nil {
+				return true, nil
+			}
+
+			return false, nil
+		})
+		if err != nil {
+			return err
+		}
+
+		steveclient, err := client.Steve.ProxyDownstream(installOptions.Cluster.ID)
+		if err != nil {
+			return err
+		}
+
+		namespaceClient := steveclient.SteveType(namespaces.NamespaceSteveType)
+
+		namespace, err := namespaceClient.ByID(CISBenchmarkNamespace)
+		if err != nil {
+			return err
+		}
+
+		err = namespaceClient.Delete(namespace)
+		if err != nil {
+			return err
+		}
+
+		adminClient, err := rancher.NewClient(client.RancherConfig.AdminToken, client.Session)
+		if err != nil {
+			return err
+		}
+
+		adminDynamicClient, err := adminClient.GetDownStreamClusterClient(installOptions.Cluster.ID)
+		if err != nil {
+			return err
+		}
+
+		adminNamespaceResource := adminDynamicClient.Resource(kubenamespaces.NamespaceGroupVersionResource).Namespace("")
+
+		watchNamespaceInterface, err := adminNamespaceResource.Watch(context.TODO(), metav1.ListOptions{
+			FieldSelector:  "metadata.name=" + CISBenchmarkNamespace,
+			TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+		})
+		if err != nil {
+			return err
+		}
+
+		return wait.WatchWait(watchNamespaceInterface, func(event watch.Event) (ready bool, err error) {
+			if event.Type == watch.Deleted {
+				return true, nil
+			}
+
+			return false, nil
+		})
+	})
+
+	err = catalogClient.InstallChart(chartInstallAction, catalog.RancherChartRepo)
+	if err != nil {
+		return err
+	}
+
+	watchAppInterface, err := catalogClient.Apps(CISBenchmarkNamespace).Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + CISBenchmarkName,
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	if err != nil {
+		return err
+	}
+
+	err = wait.WatchWait(watchAppInterface, func(event watch.Event) (ready bool, err error) {
+		app := event.Object.(*catalogv1.App)
+
+		state := app.Status.Summary.State
+		if state == string(catalogv1.StatusDeployed) {
+			return true, nil
+		}
+
+		return false, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// newCISBenchmarkChartInstallAction is a private helper function that returns chart install action with CIS benchmark and payload options.
+func newCISBenchmarkChartInstallAction(p *payloadOpts) (*types.ChartInstallAction, error) {
```

Comment:
I've got two questions here:

1. Does CIS Benchmark have any options - like Jaeger with Istio? (if yes) is it expected to have a struct for future tests, if not ignore this one, please.
2. Does this default template `ChartInstall` payload value provide support on the K3s, RKE1, and RKE2 types? So installation on any provider type is out-of-the-box?

Reply:
1. It's not expected to have a struct, correct.
2. That's correct, cluster type is agnostic here. `ChartInstall` is provided for any provider type out of the box, which is makes it pretty straightforward.